sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeter.java

```diff
@@ -145,30 +145,54 @@ void resetForTest() {
 
   @Override
   public LongCounterBuilder counterBuilder(String name) {
-    return checkValidInstrumentName(name)
-        ? new SdkLongCounter.SdkLongCounterBuilder(this, name)
-        : NOOP_METER.counterBuilder(NOOP_INSTRUMENT_NAME);
+    if (!checkValidInstrumentName(name)) {
+      return NOOP_METER.counterBuilder(NOOP_INSTRUMENT_NAME);
+    }
+    try {
+      Class.forName("io.opentelemetry.api.incubator.metrics.ExtendedLongCounter");
```

Comment:
Was fast and loose with this. Before considering merging, would refactor to call this once and cache the result.